### PR TITLE
refactor: improve fetching error messages

### PR DESCRIPTION
**Pull request by rgrinberg**
* Include location in the error message
* Do not include verbose message
* Format errors using the correct dune primitives

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>

### CODE REVIEW

File: src/dune_pkg/fetch.mli [hunk 1]
```diff
@@ -1,3 +1,3 @@
 open Stdune
 
-val fetch : OpamUrl.t -> target:Path.t -> unit Fiber.t
+val fetch : Loc.t -> OpamUrl.t -> target:Path.t -> unit Fiber.t
```
Comment (Alizter) [inline src/dune_pkg/fetch.mli:3]: What exactly is the loc meant to represent here? Is it the location in the dune file the fetch will come from? If so can you doc.

File: src/dune_pkg/fetch.ml [hunk 2]
```diff
@@ -71,22 +71,20 @@ module Fiber_job = struct
...
+      ([ Pp.textf "%s not available" (OpamUrl.to_string url) ]
+      @
       match normal with
-      | None -> "None"
-      | Some n -> Printf.sprintf "Some %S" n
-    in
-    Format.eprintf "Not available: normal %s verbose %S\n" n verbose;
-    Fiber.return ()
+      | None -> []
+      | Some normal -> [ Pp.text normal ])
```
Comment (Alizter) [inline src/dune_pkg/fetch.ml:90]: Maybe time for an `Option.to_list_map`?

**Review by Alizter — APPROVED** [review]

File: src/dune_pkg/fetch.mli (hunk 1 above)
Comment (Leonidas-from-XIV) [reply src/dune_pkg/fetch.mli:3]: Yes, I strongly assume so after reading some other PRs.

**Review by Leonidas-from-XIV — APPROVED** [review]
LGTM

File: src/dune_pkg/fetch.ml (hunk 2 above)
Comment (rgrinberg) [reply src/dune_pkg/fetch.ml:90]: Doesn't seem useful enough.

**Pull request merged**